synapse/events/spamcheck.py

```diff
@@ -26,7 +29,13 @@ def __init__(self, hs):
             pass
 
         if module is not None:
-            self.spam_checker = module(config=config)
+            # Older spam checkers don't accept the `hs` argument, so we
+            # try and detect support.
+            spam_args = inspect.getargspec(module)
```

Comment:
You'll want to use either `getfullargspec` or `signature` instead, as `getargspec` will bail if any function annotations exist.